src/cargo/sources/registry/index.rs

```diff
@@ -261,6 +267,97 @@ struct SummariesCache<'a> {
     index_version: &'a str,
 }
 
+/// A single line in the index representing a single version of a package.
+#[derive(Deserialize)]
+pub struct RegistryPackage<'a> {
```

Comment:
Non-blocking: Should this also be renamed to `IndexPackage`?

Reply:
Brilliant name! Updated.

And CI should be fixed by #12241